app/lib/.server/llm/select-context.ts

```diff
@@ -0,0 +1,222 @@
+import { generateText, type CoreTool, type GenerateTextResult, type Message } from 'ai';
+import ignore from 'ignore';
+import type { IProviderSetting } from '~/types/model';
+import { IGNORE_PATTERNS, type FileMap } from './constants';
+import { DEFAULT_MODEL, DEFAULT_PROVIDER, PROVIDER_LIST } from '~/utils/constants';
+import { createFilesContext, extractCurrentContext, extractPropertiesFromMessage, simplifyBoltActions } from './utils';
+import { createScopedLogger } from '~/utils/logger';
+import { LLMManager } from '~/lib/modules/llm/manager';
+
+// Common patterns to ignore, similar to .gitignore
+
+const ig = ignore().add(IGNORE_PATTERNS);
+const logger = createScopedLogger('select-context');
+
+export async function selectContext(props: {
```

Comment:
this is quite a huge interface. I wish there was a smaller entity which represents a single ContextManipulationStep which can be chained. These would not need some of these parameters afaics and would make it more concise

Reply:
💯

I am not sure about the order though. Changing the current codebase without creating a proper architecture for this component can result in more ... mess. 

Thus I questioned the order in which we do that. But I understand both arguments

Reply:
I understand your point, I am more open to add stuff first and periodically refactor codebase. cuz you can never start with a perfect architecture. and it will always become messy after some point as we add features that were previously unplanned. and we cannot be starting from zero every time project becomes messy.

so i am less scared of making codebase changes that might looks messy, because once it piles up a little and we starts to see a pattern, we refactor it and updated the architecture, I am open to "heavy" refactoring if needed.